File: 01_Language/01_Functions/python/basename.py
# coding: utf-8

import os


def basename(path, suffix=None):
    res = os.path.basename(os.path.normpath(path))
    if suffix is not None:
        if res.endswith(suffix):
            res = res[:-len(suffix)]
    return res


if __name__ == '__main__':
    print(basename("/etc/sudoers.d", ".d"))
    print(basename("/etc/sudoers.d"))
    print(basename("/etc/passwd"))
    print(basename("/etc/passwd?a=b"))
    print(basename("/etc/"))
    print(basename("."))
    print(basename("/"))

File: 01_Language/01_Functions/php/basename.php
<?php

echo "1) ".basename("/etc/sudoers.d", ".d").PHP_EOL;
echo "2) ".basename("/etc/sudoers.d").PHP_EOL;
echo "3) ".basename("/etc/passwd").PHP_EOL;
echo "3) ".basename("/etc/passwd?a=b").PHP_EOL;
echo "4) ".basename("/etc/").PHP_EOL;
echo "5) ".basename(".").PHP_EOL;
echo "6) ".basename("/");
